docs/docs/dev_docs/contracts/syntax/historical_access/how_to_access_history.md

```diff
@@ -0,0 +1,99 @@
+---
+title: How to access history from a smart contract
+---
+
+The Aztec Protocol uses an append-only Merkle tree to store the headers of all previous blocks in the chain as its leaves. This is known as an archive tree. You can learn more about how it works in the [concepts section](../../../../concepts/advanced/data_structures/trees.md#archive-tree).
+
+# History library
+
+The history library allows you to access any of the following at a given block height before the current height:
+s
+* [Private notes](#note-inclusion)
+* [Notes that have been nullified](#nullifier-inclusion)
+* [Notes that have not been nullified](#note-validity)
+* [Public values](#public-value-inclusion)
+* [Contracts](#contract-inclusion)
+
+Using this library, you can check that specific notes or nullifiers happened at specific blocks. This can be useful for things such as:
```

Comment:
```suggestion
Using this library, you can check that specific notes or nullifiers were part of Aztec network state at specific blocks. This can be useful for things such as:
```